src/tea_json.ml

```diff
@@ -145,13 +145,13 @@ module Decoder = struct
                     | Tea_result.Error e -> raise (ParseFail e)
                 ) in
               let emptyDict = ObjectDict.empty in
-              try Tea_result.Ok (Array.fold_right parse keys emptyDict)
+              try Tea_result.Ok (Belt.Array.reduceReverse keys emptyDict parse)
               with ParseFail e -> Tea_result.Error ("Invalid dict parsing: " ^ e)
             )
           | _ -> Tea_result.Error "Non-dict value"
       )
 
-  let field key (Decoder decoder) =
+  let field (Decoder decoder) key =
```

Comment:
This breaks the Elm api here, must not be done for the main 'Tea' namespace!  Ditto for the rest in this module!

Reply:
ok, going to change this